### PR TITLE
DD Agorithm Cleanup

### PR DESCRIPTION
**Pull request by ianna**
* Use unique_ptr to DDAlgorithm in its handler for proper memory management
* Remove unused variable
* Cleanup

### CODE REVIEW

**Comment by ianna** [comment]
please test

File: DetectorDescription/Algorithm/src/DDAlgorithmHandler.cc [hunk 1]
```diff
@@ -30,11 +20,10 @@ DDAlgorithmHandler::initialize( const std::string & algoName,
 				const DDStringArguments & sArgs,
 				const DDStringVectorArguments & svArgs )
 {
-  std::pair<std::string,std::string> algoNmNs = DDSplit(algoName);
   algoname_ = algoName;
-  algo_ = DDAlgorithmFactory::get()->create(algoname_);
-  algo_->setParent(parent);
-  algo_->initialize(nArgs,vArgs,mArgs,sArgs, svArgs);
+  algo_ = std::unique_ptr<DDAlgorithm>( DDAlgorithmFactory::get()->create( algoname_ ));
```
Comment (Dr15Jones) [inline DetectorDescription/Algorithm/src/DDAlgorithmHandler.cc]: You could do
```
algo_ = std::make_unique<DDAlgorithm>(DDAlgorithmFactory::get()->create( algoname_ ));
```

Comment (ianna) [inline DetectorDescription/Algorithm/src/DDAlgorithmHandler.cc]: I think, there is no performance improvement in this case, besides DDAlgorithm is an abstract class with pure virtual functions.

**Comment by ianna** [comment]
+1

File: DetectorDescription/Algorithm/src/DDAlgorithmHandler.cc (hunk 1 above)
Comment (Dr15Jones) [inline DetectorDescription/Algorithm/src/DDAlgorithmHandler.cc]: The change isn't about performance. It is the recommended form to use from C++ style guides. If an exception happens during the construction phase, using the function will properly clean up everything.

Comment (ianna) [inline DetectorDescription/Algorithm/src/DDAlgorithmHandler.cc]: ok, but I'd need to write more code as make_unique would try to new DDAlgorithm

Comment (Dr15Jones) [inline DetectorDescription/Algorithm/src/DDAlgorithmHandler.cc]: You are right. I should only review code after more caffeine. I withdraw the suggestion.

**Comment by davidlange6** [comment]
+1

**Pull request merged**